statefun-e2e-tests/statefun-smoke-e2e-embedded/src/main/java/org/apache/flink/statefun/e2e/smoke/EmbeddedModule.java

```diff
@@ -25,7 +25,7 @@
 
 @AutoService(StatefulFunctionModule.class)
```

Comment:
I realized that the Java service provider configuration file generated by `AutoService` in the fat uber jar (i.e., the `META-INF/services/org.apache.flink.statefun.sdk.spi.StatefulFunctionModule` file you'll see in the built jar) does not correctly list `EmbeddedModule` as an implementation. It only lists the `DriverModule`.

This should probably be addressed with a resource transformer via the Maven shade plugin. Both `EmbeddedModule` and `DriverModule` needs to be listed in the service provider configuration file, so that the StateFun runtime picks both up as a registered module.

The only reason this "coincidentally" works is because you are adding two jars in the `Dockerfile` right now, with the non-uber jar containing a service provider file for `EmbeddedModule`, and the uber jar containing a service provider for `DriverModule`. Strictly speaking, we only need to add the uber jar, which should contain a service provider file listing both modules.

Reply:
OK. That wasn't intentional ;)